R/get_delay_sample_fn.R

```diff
@@ -35,26 +36,59 @@ get_delay_sample_fn <- function(linelist, verbose = FALSE, samples = 1) {
   ## Maximum allowed delay
   max_delay <- max(confirmation_delays) + 1
 
-  fit <- EpiNow::dist_fit(confirmation_delays, samples = samples, dist = "exp")
+  # Fit gamma and exponential models
+  fit_exp <- EpiNow::dist_fit(confirmation_delays, samples = samples, dist = "exp")
+  fit_gam <- EpiNow::dist_fit(confirmation_delays, samples = samples, dist = "gamma")
+  # Extract log likelihoods
+  log_lik_exp <- loo::extract_log_lik(fit_exp, merge_chains = FALSE)
+  log_lik_gam <- loo::extract_log_lik(fit_gam, merge_chains = FALSE)
+  # Calculate relative efficiencies
+  rel_exp <- loo::relative_eff(exp(log_lik_exp))
+  rel_gam <- loo::relative_eff(exp(log_lik_gam))
+  # Estimate looic
+  loo_exp <- loo::loo(log_lik_exp, r_eff = rel_exp, cores = 2)
```

Comment:
Need to default to 1 core as everything is run in parallel already so this will cause some problems (I think)